boards/itsybitsy_m4/Cargo.toml

```diff
@@ -18,10 +18,11 @@ apa102-spi = "0.3"
 smart-leds = "0.3"
 
 [dependencies.cortex-m-rt]
-version = "0.6.12"
+version = "0.7"
 optional = true
 
 [dependencies.atsamd-hal]
+path="../../hal"
```

Comment:
This would make it tier one, which I don't think we want to do

Reply:
And after removing this, I think you might need to revert the `cortex-m` change :/

Reply:
The updated bsp relies on some changes in the hal:
- pac dependency atsamd51g 0.11.0 (the currently published hal depends on the 0.10.0).
- Addition of the UndocIoSet2 is required for the setting up the `spi_master` function.
- Some changes in the spi v2 api are required, like the addition of spi::Duplex.

There are two ways of resolving this I think
First one is:
- Close the PR
- Make a separate just for adding de UndocIoSet
- Wait for hal version bump
- Open new PR with the bsp updates
Second One:
- Have the bsp be Tier 1 but not listing it as that. 
- Make it Tier 2 as soon as new version of the hal is released (change the hal dependency path on the version release commit)

I think option 2 is similar to what the `circuit_playground_express` bsp did on #530